pkg/proxy/ipvs/proxier.go

```diff
@@ -1944,59 +1945,58 @@ func (proxier *Proxier) deleteEndpointConnections(connectionMap []proxy.ServiceE
 	}
 }
 
-func (proxier *Proxier) syncService(svcName string, vs *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) error {
-	appliedVirtualServer, _ := proxier.ipvs.GetVirtualServer(vs)
-	if appliedVirtualServer == nil || !appliedVirtualServer.Equal(vs) {
-		if appliedVirtualServer == nil {
+func (proxier *Proxier) syncService(svcName string, updatedVS, currentVS *utilipvs.VirtualServer, bindAddr bool, bindedAddresses sets.String) (*utilipvs.VirtualServer, error) {
```

Comment:
I think it would be better for `syncService` to pass the entire list of `currentVS` in here, or store `currentVS` as a field in the `Proxier` struct. We should use a read/write mutex lock for checking entries in there as well to avoid race conditions.

Reply:
The lock is maybe not required since `syncProxyRules` is already locked, but I think store `currentVS` into a struct field would be useful, that way we can fetch from it in any method as opposed to passing it in every function that needs it.

Reply:
Sorry for taking so long to revisit this PR. I have updated it with these suggestions moving the `currentIPVSServices` map to the `Proxier` struct.